File: abstract-factory/java/src/shop/DirectSaleShop.java
package src.shop;

// 具体产品类，实现抽象产品接口
public class DirectSaleShop implements Shop {
  @Override
  public void greetings() {
     System.out.println("DirectSaleShop::greetings().");
  }

  public void welcome() {
     System.out.println("DirectSaleShop::welcome().");
  }
}

File: abstract-factory/python/src/shop/DirectSaleShop.py
# -*- coding: utf-8 -*-
"""
@author: jarry
"""
from src.shop.Shop import Shop


class DirectSaleShop(Shop):

    def greetings(self):
        print('DirectSaleShop::greetings')

    def welcome(self):
        print('DirectSaleShop::welcome')
